<a href="https://colab.research.google.com/github/Drashti-Naik/colab-python/blob/main/Sampling_Distribution.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Define population and sample population

Introduce the concept of sampling and creating a sample distribution of the population.

Identify the relationship between mean and SD vs mean and SD of sample distribution.


In [2]:
import plotly.figure_factory as ff
import plotly.graph_objects as go
import statistics
import random
import pandas as pd

df = pd.read_csv("https://raw.githubusercontent.com/whitehatjr/Sampling-distribution/master/data.csv")
df.head()
#df.tail()
data=df["temp"].tolist()
print(data)


[29, 29, 41, 41, 31, 31, 29, 29, 29, 29, 42, 42, 29, 29, 29, 29, 41, 41, 29, 29, 42, 42, 29, 29, 41, 41, 29, 29, 29, 29, 29, 29, 42, 42, 29, 29, 41, 41, 29, 42, 30, 29, 42, 30, 30, 29, 29, 29, 29, 41, 30, 30, 42, 42, 29, 29, 31, 29, 29, 29, 29, 29, 29, 29, 41, 30, 30, 29, 29, 29, 29, 29, 29, 29, 29, 41, 41, 29, 29, 41, 41, 29, 29, 29, 29, 41, 41, 29, 29, 41, 41, 29, 29, 41, 41, 29, 29, 40, 40, 29, 29, 29, 29, 30, 30, 29, 29, 40, 40, 30, 30, 29, 29, 40, 40, 29, 29, 29, 29, 40, 40, 29, 29, 40, 40, 29, 41, 29, 41, 29, 29, 29, 29, 29, 40, 29, 40, 29, 29, 29, 41, 41, 29, 29, 40, 40, 29, 29, 29, 29, 29, 29, 29, 29, 29, 29, 39, 29, 29, 29, 29, 39, 38, 39, 29, 40, 29, 29, 39, 39, 29, 29, 29, 39, 29, 39, 29, 39, 29, 39, 39, 29, 29, 29, 29, 29, 29, 29, 29, 39, 39, 29, 29, 29, 29, 29, 29, 29, 29, 29, 29, 29, 39, 29, 29, 29, 39, 29, 29, 29, 29, 38, 39, 29, 39, 29, 39, 29, 38, 39, 29, 30, 30, 29, 29, 38, 30, 29, 30, 30, 29, 29, 29, 38, 38, 29, 29, 38, 38, 29, 29, 38, 38, 29, 29, 29, 29, 29, 29, 29,

In [5]:
#find mean of raw data("population")
mean_population=statistics.mean(data)
print("population_mean:",mean_population)

population_mean: 35.05393111079237


In [6]:
#SD of population
std_population = statistics.stdev(data)
print("population_std:",std_population)

population_std: 5.699825337585306


In [8]:
ff.create_distplot([data],["Temp"],show_hist=False)

In [10]:
print(data)

[29, 29, 41, 41, 31, 31, 29, 29, 29, 29, 42, 42, 29, 29, 29, 29, 41, 41, 29, 29, 42, 42, 29, 29, 41, 41, 29, 29, 29, 29, 29, 29, 42, 42, 29, 29, 41, 41, 29, 42, 30, 29, 42, 30, 30, 29, 29, 29, 29, 41, 30, 30, 42, 42, 29, 29, 31, 29, 29, 29, 29, 29, 29, 29, 41, 30, 30, 29, 29, 29, 29, 29, 29, 29, 29, 41, 41, 29, 29, 41, 41, 29, 29, 29, 29, 41, 41, 29, 29, 41, 41, 29, 29, 41, 41, 29, 29, 40, 40, 29, 29, 29, 29, 30, 30, 29, 29, 40, 40, 30, 30, 29, 29, 40, 40, 29, 29, 29, 29, 40, 40, 29, 29, 40, 40, 29, 41, 29, 41, 29, 29, 29, 29, 29, 40, 29, 40, 29, 29, 29, 41, 41, 29, 29, 40, 40, 29, 29, 29, 29, 29, 29, 29, 29, 29, 29, 39, 29, 29, 29, 29, 39, 38, 39, 29, 40, 29, 29, 39, 39, 29, 29, 29, 39, 29, 39, 29, 39, 29, 39, 39, 29, 29, 29, 29, 29, 29, 29, 29, 39, 39, 29, 29, 29, 29, 29, 29, 29, 29, 29, 29, 29, 39, 29, 29, 29, 39, 29, 29, 29, 29, 38, 39, 29, 39, 29, 39, 29, 38, 39, 29, 30, 30, 29, 29, 38, 30, 29, 30, 30, 29, 29, 29, 38, 38, 29, 29, 38, 38, 29, 29, 38, 38, 29, 29, 29, 29, 29, 29, 29,

100 random data set from row data

In [9]:
import random
dataset = []
for i in range(0,100):
  random_index=random.randint(0,len(data))
  value=data[random_index]
  dataset.append(value)
mean=statistics.mean(dataset)
std = statistics.stdev(dataset)

print("mean of sample:",mean)
print("SD of sample:",std)

mean of sample: 34.43
SD of sample: 5.466620472827741


Now lets do it 1000 times. let us collect 1000 random sample of 100 data points from the population. find mean and plot the distribution of all sample means.

In [13]:
def random_set_ofmean(counter):
  dataset=[]
  for i in range(0,counter):
     random_index=random.randint(0,len(data)-1)
     value=data[random_index]
     dataset.append(value)
  mean=statistics.mean(dataset)
  return mean

def show_fig(mean_list):
  df=mean_list
  mean=statistics.mean(df)
  fig = ff.create_distplot([df],["Temp"],show_hist=False)
  fig.add_trace(go.Scatter(x=[mean,mean],y=[0,1],mode="lines",name="MEAN"))
  fig.show()

def setup():
  mean_list=[]
  for i in range(0,1000):
    set_of_means=random_set_ofmean(100)
    mean_list.append(set_of_means)
  show_fig(mean_list)

  mean=statistics.mean(mean_list)
  std = statistics.stdev(mean_list)
  print("Mean of sampling Distribution:",mean)
  print("std of sampling Distribution:",std)


setup()




Mean of sampling Distribution: 35.0272
std of sampling Distribution: 0.5593015091863941


This kind of distribution is called sampling distribution.And provided we take sufficient numbe of samples,the sampling mean distribution for any kind of data will always be normal distribution.

In [14]:
std_population

5.699825337585306

What is the SD of sampling mean distribution? Do you observer any relation with the population SD?

**SD of sampling mean = 0.5593**

**SD of sampling mean = 1/10 of population of SD**





Here, the number of data in each sample was 100.so, we found the SD of sampling mean distribution = 1/10 * SD of population


**mean of sampling disttibution = population mean**

**SD of sampling mean distribution = SD of population / sqrt(number of data in each sample)**